src/native/windows/apps/prunsrv/prunsrv.c

```diff
@@ -951,15 +970,34 @@ static BOOL docmdStopService(LPAPXCMDLINE lpCmdline)
                                NULL);
         if (!rv) {
             /* Wait for the timeout if any */
-            int  timeout     = SO_STOPTIMEOUT;
-            if (timeout) {
-                int i;
-                for (i = 0; i < timeout; i++) {
-                    rv = apxServiceCheckStop(hService);
-                    apxLogWrite(APXLOG_MARK_DEBUG "apxServiceCheck returns %d.", rv);
-                    if (rv)
-                        break;
-                }
+            int timeout     = SO_STOPTIMEOUT;
+            int waited      = waitedSinceStopCmd();
+            int i;
+            if (timeout <= 0) {
+                /* waiting for ever doesn't look OK here */
+                timeout = ONE_MINUTE_SEC;
+            }
```

Comment:
If we aren't ever going to allow an infinite timeout, why let users create one in the first place? Should we reject any attempt to create such a service? If we go that route, we'll need to allow for existing services that do have an infinite timeout. Should be log a warning that we are ignoring the timeout?

Reply:
OK I have fixed to reject any attempt to create such a service.